src/cpp/session/resources/grid/gridviewer.js

```diff
@@ -611,6 +624,83 @@ var createFilterUI = function(idx, col) {
   return host;
 };
 
+var openColumnTypeChooserUI = function(idx, col, ele, onDismiss) {
+  invokeFilterPopup(ele, function(popup) {
+    var columnTypes = [
+      "logical", "integer", "double", "character", "date", "datetime", "numeric" 
```

Comment:
I based the supported types from `vignette("column-types")`; however, those seem only the auto-detectable ones. I'll add all the types I find under `readr::col_*`